src/routes/docs/tutorials/nextjs/step-4/+page.markdoc

```diff
@@ -0,0 +1,217 @@
+
+```js
+"use client";
+
+import Link from "next/link";
+import toast from "react-hot-toast";
+import { useRouter } from "next/navigation";
+import { ID } from "appwrite";
+import { useCookie } from "next-cookie";
+
+import { account } from "@/config/appwrite";
+
+const SignUp = (props) => {
+  const router = useRouter();
+  const cookie = useCookie(props.cookie);
+
+  return (
+    <section className="container u-flex u-main-center u-cross-center u-full-screen-height">
+      <div className="u-max-width-350	u-width-full-line u-padding-8">
+        <h1 className="heading-level-1 font-bold">Sign Up</h1>
+        <p className="body-text-1 u-bold u-padding-block-16">
+          Already have an account?{" "}
+          <Link href="/login" className="u-cursor-pointer u-underline">
+            Login
+          </Link>
+        </p>
+        <form onSubmit={signup}>
+          <div>
+            <label className="label">Name</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Name"
+                name="name"
+                type="text"
+                required
+              />
+            </div>
+          </div>
+          <div>
+            <label className="label">Email</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Email"
+                name="email"
+                type="email"
+                required
+              />
+            </div>
+          </div>
+          <div>
+            <label className="label">Password</label>
+            <div className="input-text-wrapper">
+              <input
+                className="input-text"
+                placeholder="Password"
+                name="password"
+                type="password"
+                required
+              />
+            </div>
+          </div>
+          <div className="u-padding-block-16">
+            <button type="submit" className="button">
+              Sign Up
+            </button>
+          </div>
+        </form>
+      </div>
+    </section>
+  );
+};
+
+export default SignUp;
+```
+
+Now let's add the signup function. We will be using Appwrite's `create` method to create a new user.
+
+
+```js
+// /src/app/(pages)/signup/page.js
+const signup = async (e) => {
+  e.preventDefault();
+  const form = e.target;
+  const formData = new FormData(form);
+  const email = formData.get("email");
+  const password = formData.get("password");
+  const name = formData.get("name");
+  if (!email || !password || !name) {
+    toast.error("Please fill in all the fields");
+    return;
+  }
+  try {
+    await account.create(ID.unique(), email, password, name);
+    await account.createEmailSession(email, password);
+    const jwt = await account.createJWT();
```

Comment:
I think there's an issue here with JWTs expiring every 30 minutes 🤔

Reply:
it's expiring even early and I can't find a way to resolve this issue.

Reply:
@Mridul2820 We may have to wait for proper SSR support from our side 😬

When we accept this, we'll hold off on releasing it until we have proper SSR support.